SS14.Server/BaseServer.cs

```
@@ -255,8 +260,14 @@ public void MainLoop()
             _mainLoop.Tick += (sender, args) => Update(args.DeltaSeconds);
 
             // set GameLoop.Running to false to return from this function.
-            _mainLoop.Run();
-
+            try
+            {
+                _mainLoop.Run();
+            }
+            catch (Exception exp)
+            {
+                runtimeLog.AddException(exp);
```

Right now this just silently eats the exceptions with no way to be aware of them. The general idea behind the system is definitely a step in the right direction but taking out the logging is not a good idea.